android/src/main/java/app/notifee/core/utility/ExtendedListenableFuture.java

```diff
@@ -0,0 +1,74 @@
+package app.notifee.core.utility;
+
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class ExtendedListenableFuture<T> implements ListenableFuture<T> {
```

Comment:
This one (and Callbackable) could use at least a class doc to explain why they are necessary, how to use appropriately

Reply:
okay

Reply:
I added docs for them, can you check if it's fine ?
And I am really not sure If I am doing it properly at all 😅 I'm not that good with Java, and honestly I have memory leak concerns about how I have used `ExtendedListenableFuture` , but I have no idea how to check if the way I have used it is appropriate or not